types/puppeteer/index.d.ts

```diff
@@ -667,7 +667,7 @@ export interface FrameBase {
    */
   $eval(
     selector: string,
-    fn: (element: ElementHandle | null, ...args: any[]) => any,
+    fn: (element: Element | null, ...args: any[]) => any,
```

Comment:
The function will only receive a value of type `Element`, it will never receive a `null`. When the `selector` cannot be found it will throw an error:
 `Error: failed to find element matching selector ".selector"`.

The rest arguments can be a `ElementHandle` or anything serializable, thus `ElementHandle|any`.

Last thing; the return value is always of type `Promise<any>`, as it is always a Promise and we don't know the context type upfront.

So `fn`, which should be called `pageFuncton` by the documentation, should be typed like this:

`pageFunction: (element: Element, ...args: (ElementHandle|any)[]) => Promise<any>`

Reply:
Thanks for feedback. 

According to document:

> _**If**_ pageFunction returns a Promise, then page.$eval would wait for the promise to resolve and return its value.

And the example of document:
```typescript
const searchValue = await page.$eval('#search', el => el.value);
const preloadHref = await page.$eval('link[rel=preload]', el => el.href);
const html = await page.$eval('.main-container', e => e.outerHTML);
```
It seems that the return value of pageFunction could be but not necessarily be a Promise.

Reply:
> It seems that the return value of pageFunction could be but not necessarily be a Promise.

That's my understanding as well, so `=> any` is correct